### ISSUE TITLE
statement triggers

### ISSUE THREAD

**Issue by ryanhiebert** [jyveapp/django-pgtrigger#2]
It looks like this package is doing triggers `FOR EACH ROW`. I'm wanting to create a `FOR EACH STATEMENT` trigger that updates a materialized view when a particular table is changed. It's a very low-write table, that's read using an annoyingly complicated RECURSIVE query. Is this something that this package is interested in supporting?

**Comment by wesleykendall** [comment]
@ryanhiebert absolutely, I believe this will be a trivial addition, although you will still need to write the core pgpsql function that updates your matview. The trigger definition would look something like this:

Assuming you have access to the model you want to track:

```
@pgtrigger.register(
    pgtrigger.Trigger(
        when=pgtrigger.After,
        operation=(pgtrigger.Insert | pgtrigger.Update | pgtrigger.Delete | pgrigger.Truncate),  # Any operation
        level=pgtrigger.Statement,  # A new addition to the interface. The "level" of trigger
        func='Custom SQL statement(s) that will be inside your trigger function'
    )
)
class ModelBeingTracked(models.Model):
    ...
```

I haven't used statement-level triggers yet, but it looks like the only addition needed is the concept of the "level" of the trigger. I will probably default it to "Row" to preserve backwards compatibility.

Would this work for your use case?

**Comment by ryanhiebert** [comment]
Yeah, that looks like it would work great!

**Comment by ryanhiebert** [comment]
Right now I'm using a row-level trigger, because I can live with it for the time being, but it unlocks some performance improvements if I can make it statement level instead.

**Comment by wesleykendall** [comment]
Addressed in https://github.com/jyveapp/django-pgtrigger/pull/3, which also adds support for the REFERENCING construct in statement-level triggers in case people want access to transition tables

**Comment by wesleykendall** [comment]
@tomage this is addressed in v1.1.0 now (deployed!). Feel free to close this one
